gateway/ln-gateway/src/lightning/ldk.rs

```diff
@@ -54,6 +62,15 @@ pub struct GatewayLdkClient {
     /// The HTLC stream, until it is taken by calling
     /// `ILnRpcClient::route_htlcs`.
     htlc_stream_receiver_or: Option<tokio::sync::mpsc::Receiver<InterceptPaymentRequest>>,
+
+    /// The set of outgoing payments, by payment hash, that are currently
+    /// awaiting completion. This is used to allow `Self::handle_next_event()`
+    /// to notify `Self::pay()` when a payment completes.
+    payment_subscriptions: Arc<
+        Mutex<
+            HashMap<bitcoin30::hashes::sha256::Hash, tokio::sync::oneshot::Sender<PaymentOutcome>>,
```

Comment:
nit: I prefer `BTreeMap`

Reply:
It might also be worth making it a type def, given the length of the type

Reply:
Done and done